scripts/deploy_bug.py

```diff
@@ -66,57 +72,63 @@ def main(argv):
 
     # Convert to int and drop any that don't
     tags = [int(tag) for tag in tags if tag.isdigit()]
+    return tags
+
+
+def main(argv):
+    # Fetch tags as numbers (e.g. 309)
+    tags = get_tags()
 
     # The current tag is the max of that
     current_tag = max(tags)
 
     # The next tag is current_tag + 1
     next_tag = current_tag + 1
 
+    # Figure out the current version
+    sha = fetch_current_revision(HOST)
+
+    # Get the commits between the currently deployed version and what's in
+    # master tip and if there's nothing to deploy, then we're done!
+    resp = fetch_history_from_github('mozilla-services', 'socorro', sha)
+    if resp['status'] != 'ahead':
+        print('Nothing to deploy!')
+        return
+
+    commits = resp['commits']
+
     print('summary:')
     print('socorro deploy: %s' % next_tag)
     print()
-
     print('description:')
     print()
-
     print('We want to do a Socorro -prod deploy today tagged %s.' % next_tag)
     print()
-
     print('It consists of the following:')
     print()
+    print('(current tag: %s - %s)' % (current_tag, sha[:8]))
 
-    # Figure out and print the version
+    # Print the commits out skipping merge commits
+    for commit in commits:
+        if len(commit['parents']) > 1:
+            continue
 
-    # NOTE(willkg): This is the current infrastructure--the new infrastructure
-    # uses the /__version__ endpoint. We'll need to update this when we switch
-    # over.
-    resp = fetch('https://crash-stats.mozilla.com/api/Status/')
-    sha = resp['socorro_revision']
+        print('%s: %s' % (
+            commit['sha'][:8],
+            commit['commit']['message'].splitlines()[0][:80]
+        ))
 
-    # Figure out and print everything that's going out
-    print('(current tag: %s - %s)' % (current_tag, sha[:8]))
-    resp = fetch_history_from_github('mozilla-services', 'socorro', sha)
-    if resp['status'] == 'ahead':
-        # Drop all the merge commits
-        commits = [commit for commit in resp['commits'] if len(commit['parents']) == 1]
-
-        # Print the commits out
-        for commit in commits:
-            print_row(
-                commit['sha'][:8],
-                commit['commit']['message'].splitlines()[0][:80]
-            )
     print('(next tag: %s - %s)' % (next_tag, commits[-1]['sha'][:8]))
     print()
 
-    # Print post-deploy steps
+    # Print all possible post-deploy steps--we winnow the unnecessary ones when
```

Comment:
I like the word "winnow". I want to use it more often. I'm going to make this a 2018h1 goal.